File: backend/app/schemas/content.py
from typing import List, Literal, Optional

from pydantic import BaseModel


class ContentItem(BaseModel):
    id: str
    type: Literal["sentence", "dialog", "word"]
    tags: List[str] = []
    level: Optional[str] = None
    text: str
    audio_url: Optional[str] = None
    segments: List[str] = []
    segments_audio: Optional[List[str]] = None


class ContentListResponse(BaseModel):
    items: List[ContentItem]
    total: int

File: miniprogram/services/content.ts
import { request } from './request';

export interface ContentItem {
  id: string;
  type: string;
  text: string;
  audio_url?: string;
  segments?: string[];
  segments_audio?: string[];
  level?: string;
  tags?: string[];
}

interface ContentListResponse {
  items: ContentItem[];
  total: number;
}

export function list(params: Record<string, any> = {}): Promise<ContentListResponse> {
  return request.get<ContentListResponse>('/content', params);
}

export function get(id: string): Promise<ContentItem> {
  return request.get<ContentItem>(`/content/${id}`);
}
